hotness/consumers.py

```diff
@@ -796,140 +733,230 @@ def get_dist_tag(self):
-            for old_task_id, (old_bz, old_state, old_version, old_package) in \
-                    six.iteritems(self.old_triggered_task_ids):
-                if int(old_bz.bug_id) == int(bz.bug_id):
-                    if new_state == 'CANCELED':
-                        self.new_triggered_task_ids.pop(task_id)
-                        self.old_triggered_task_ids.pop(old_task_id)
-                        bug = None
-                        rh_stuff = None
-                    # Koji build was successful and we can compare both packages
-                    elif new_state == 'CLOSED':
-                        if old_state is not None:
-                            # We do not want to call rebase-helper if old koji was cancelled
-                            # remove both task from triggers
-                            if old_state == 'CLOSED':
-                                rh_stuff = self.buildsys.rebase_helper_checkers(new_version,
-                                                                                str(old_task_id),
-                                                                                str(task_id),
-                                                                                tmp)
-                                bug = bz
-                                text = text % (new_state, task_id)
-                                self.bugzilla.follow_up(text, bug)
-                            # Old sources failed or cancelled. We do not have nothing to do
-                            elif old_state in ('FAILED', 'CANCELED'):
-                                bug = None
-                                rh_stuff = None
-                            self.old_triggered_task_ids.pop(old_task_id)
-                            self.new_triggered_task_ids.pop(task_id)
-                        # Old sources has not been finished yet
-                        else:
-                            bug = None
-                            rh_stuff = None
-                    # Koji build failed and we have to attach build logs from rebase-helper
-                    else:
-                        if old_state is not None and old_state not in ('FAILED', 'CANCELED'):
-                            rh_stuff = self.buildsys.rebase_helper_checkers(new_version,
-                                                                            str(old_task_id),
-                                                                            str(task_id),
-                                                                            tmp)
-                            bug = bz
-                            self.old_triggered_task_ids.pop(old_task_id)
-                            self.new_triggered_task_ids.pop(task_id)
-                            text = text % (new_state, task_id)
-                            self.bugzilla.follow_up(text, bug)
+
+        old_task_id, new_task_id = self._find_tasks(task_id)
+
+        if old_task_id is None or new_task_id is None:
```

Comment:
This could also be ``if None in (old_task_id, new_task_id)``, but that's a nitpick and I don't mind either way.